src/www/ui/api/Controllers/GroupController.php

```diff
@@ -92,4 +93,71 @@ public function createGroup($request, $response, $args)
     }
     return $response->withJson($returnVal->getArray(), $returnVal->getCode());
   }
+
+  /**
+   * Change user permission
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function changeUserPermission($request, $response, $args)
+  {
+
+    // Extract all  prerequisites (dbManager , user_pk , new_permission , gropu_pk ) for this functionality
+    $dbManager = $this->dbHelper->getDbManager();
+    $user_pk = intval($args['userId']);
+    $newperm = intval(json_decode($request->getBody()->getContents(), true)['perm']);
+    $group_pk = intval($args['id']);
+
+
+    // Validate arguments
+    if (!$this->dbHelper->doesIdExist("groups", "group_pk", $group_pk)) {
+      $info = new Info(404, "GroupId doesn't exist", InfoType::ERROR);
+    } else if (!$this->dbHelper->doesIdExist("users", "user_pk", $user_pk)) {
+      $info = new Info(404, "UserId doesn't exist", InfoType::ERROR);
+    } else if ($newperm < 0) {
+      $info = new Info(400, "ERROR - permission can not be negative", InfoType::ERROR);
+    } else if ($newperm > 2) {
+      $info = new Info(400, "ERROR - permission can not be greater than 2", InfoType::ERROR);
+    } else {
+
+      // Check if the relation already exists and retrieve the PK.
+      // IF not create a new relation and get PK.
+
+      $dbManager->prepare($stmt = __METHOD__ . ".getByGroupAndUser",
+        "SELECT group_user_member_pk FROM group_user_member WHERE group_fk=$1 AND user_fk=$2");
+      $fetchResult = $dbManager->execute($stmt, array($group_pk, $user_pk));
+      $fetchResult = $dbManager->fetchAll($fetchResult);
+      $dbManager->freeResult($fetchResult);
+
+      $group_user_member_pk = $fetchResult[0]['group_user_member_pk'];
+
+      if (empty($fetchResult)) {
+        // insert a new record
+        $dbManager->prepare($stmt = __METHOD__ . ".insertGUP",
+          "insert into group_user_member (group_fk, user_fk, group_perm) values ($1,$2,$3)");
+        $result = $dbManager->execute($stmt, array($group_pk, $user_pk, $newperm));
+        $dbManager->freeResult($result);
+
+
+        $dbManager->prepare($stmt = __METHOD__ . ".getByGroupAndUser",
+          "SELECT group_user_member_pk FROM group_user_member WHERE group_fk=$1 AND user_fk=$2");
+        $fetchResult = $dbManager->execute($stmt, array($group_pk, $user_pk));
+        $fetchResult = $dbManager->fetchAll($fetchResult);
+        $dbManager->freeResult($fetchResult);
+
+        $group_user_member_pk = $fetchResult[0]['group_user_member_pk'];
+      }
+
+      // Call the function to update the permission
+      $adminGroupUsers = new AdminGroupUsers();
```

Comment:
Please use `getPlugin()` to get the object.
```suggestion
      $adminGroupUsers = $this->restHelper->getPlugin('group_manage_users');
```

Make sure to check for `$adminGroupUsers == null` in case user does not have required permissions.